pkg/controller/networkpolicy/controller_querier_replier.go

```diff
@@ -20,119 +20,122 @@ package networkpolicy
 }
 
 // NewNetworkPolicyController returns a new *NetworkPolicyController.
-func NewEndpointQueryReplier(
-	addressGroupStore storage.Interface,
-	appliedToGroupStore storage.Interface,
-	internalNetworkPolicyStore storage.Interface,
-	podInformer coreinformers.PodInformer) *EndpointQueryReplier {
+func NewEndpointQueryReplier(networkPolicyController *NetworkPolicyController) *EndpointQueryReplier {
 	n := &EndpointQueryReplier{
-		addressGroupStore:          addressGroupStore,
-		appliedToGroupStore:        appliedToGroupStore,
-		internalNetworkPolicyStore: internalNetworkPolicyStore,
-		podInformer: podInformer,
+		networkPolicyController: networkPolicyController,
 	}
 	return n
 }
 
 //Query functions
-func (eq EndpointQueryReplier) QueryNetworkPolicies(namespace string, podName string) *EndpointQueryResponse {
+func (eq EndpointQueryReplier) QueryNetworkPolicies(namespace string, podName string) (*EndpointQueryResponse, error) {
 	// check if namespace and podName select an existing pod
-	_, err := eq.podInformer.Lister().Pods(namespace).Get(podName)
-	// TODO: how to make sure that I handle correct error
+	_, err := eq.networkPolicyController.podInformer.Lister().Pods(namespace).Get(podName)
 	if err != nil {
 		return &EndpointQueryResponse{
 			Endpoints: nil,
-			Error:     err,
-		}
+		}, err
 	}
-	// grab list of all policies from internalNetworkPolicyStore
-	internalPolicies := eq.internalNetworkPolicyStore.List()
 	// create network policies categories
-	applied, _, _ := make([]antreatypes.NetworkPolicy, 0), make([]antreatypes.NetworkPolicy, 0),
-		make([]antreatypes.NetworkPolicy, 0)
-	// filter all policies into appropriate groups
-	for _, policy := range internalPolicies {
-		antreaPolicy := policy.(*antreatypes.NetworkPolicy)
-		println(antreaPolicy.Name)
-		for _, key := range antreaPolicy.AppliedToGroups {
-			// Check if policy is applied to endpoint
-			//TODO: what is this boolean. what is this error?
-			appliedToGroupInterface, _, _ := eq.appliedToGroupStore.Get(key)
-			appliedToGroup := appliedToGroupInterface.(*antreatypes.AppliedToGroup)
-			// if appliedToGroup selects pod in namespace, append policy to applied category
-			for _, podSet := range appliedToGroup.PodsByNode {
-				for _, member := range podSet {
-					trialPodName, trialNamespace := member.Pod.Name, member.Pod.Namespace
-					println(trialPodName)
-					if podName == trialPodName && namespace == trialNamespace {
-						applied = append(applied, *policy.(*antreatypes.NetworkPolicy))
+	applied, ingress, egress := make([]*antreatypes.NetworkPolicy, 0), make([]*antreatypes.NetworkPolicy, 0),
+		make([]*antreatypes.NetworkPolicy, 0)
+	// get all appliedToGroups using pod index, then get applied policies using appliedToGroup
+	appliedToGroups, err := eq.networkPolicyController.appliedToGroupStore.GetByIndex(store.PodIndex, podName + "/" + namespace)
+	if err != nil {
+		return &EndpointQueryResponse{
```

Comment:
when returning an error don't bother building a `EndpointQueryResponse` object, it's too verbose. Return `nil, err`.